### ISSUE TITLE
[5.x.x] Type is strange for storage EventTask

### ISSUE THREAD

**Issue by tkow** [invertase/react-native-firebase#2302]
## Issue

https://github.com/invertase/react-native-firebase/blob/v5.x.x/src/index.d.ts#L239-L240

It may be one nit, but we can't use TaskEvent type in [this example](https://rnfirebase.io/docs/v5.x.x/storage/reference/Reference) without unreasonable cast using typescript.

Describe your issue here

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
 OUTPUT GOES HERE
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `e.g. 5.4.3`
- **`Firebase` module(s) you're using that has the issue:**
  - `e.g. Instance ID`
- **Are you using `TypeScript`?**
  - `Y/N`
  
</p>
</details>

---

Think `react-native-firebase` is great? Please consider supporting all of the project maintainers and contributors by donating via our [Open Collective](https://opencollective.com/react-native-firebase/donate) where all contributors can submit expenses. [[Learn More]](https://invertase.io/oss/react-native-firebase/contributing/donations-expenses)

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by mikehardy** [comment]
I use typescript as well, though I haven’t used storage, so I understand frustrating typings but just haven’t hit this myself. If you could check the typing vs v6, I’d be okay with a type change in v5 as long as it was heading towards the same type in v6 (even if v6 needs repair as well). That way we’d take a break maybe (for a v5.6.x) if it couldn’t be backwards compatible, but it wouldn’t bite twice. Propose a PR and we can see?

**Comment by tkow** [comment]
This is just v5 bug as I see it and not breaking compatibility.
In v6 it is simpler definition and doesn't have bug.

https://github.com/invertase/react-native-firebase/blob/master/packages/storage/lib/index.d.ts#L152

So, this PR should not influence most users unless force cast it correctly.

**Comment by mikehardy** [comment]
Closed with #2303
